### PR TITLE
Update filesystems tests for JeOS

### PR DESCRIPTION
**Pull request by mloviska**
- Related ticket: [[qac][jeos] snapper_undochange does not boot to to emergency shell after dropping of specific XEN pv image](https://progress.opensuse.org/issues/66577)
- Verification runs:
  * [sle-15-SP1-JeOS-for-XEN-x86_64-Build37.7.30](http://kepler.suse.cz/tests/1641#step/firstrun/1)
  * [sle-15-SP3-JeOS-for-kvm-and-xen-x86_64-Build20.3](http://kepler.suse.cz/tests/1649#step/snapper_undochange/1)
  * [sle-15-SP2-JeOS-for-kvm-and-xen-x86_64-Build15.33](http://kepler.suse.cz/tests/1652#step/snapper_undochange/21)
  * [Tumbleweed-JeOS-for-kvm-and-xen-x86_64-Build20200909](http://kepler.suse.cz/tests/1646#step/firstrun/1)
  * [15.1-JeOS-for-kvm-and-xen-x86_64-Build8.12.12](http://kepler.suse.cz/tests/1651#step/snapper_undochange/21)
  * [15.2-JeOS-for-kvm-and-xen-x86_64-Build31.165](http://kepler.suse.cz/tests/1652#step/snapper_undochange/21)

Additional test variables for oS:
``EXTRATEST=prepare,filesystem NUMDISKS=2 SNAPPER_NODBUS=1 INSTALLONLY=0 SERIAL_CONSOLE=1``

### CODE REVIEW

File: tests/jeos/firstrun.pm [hunk 1]
```diff
@@ -166,6 +166,7 @@ sub run {
 
     # openSUSE JeOS has SWAP mounted as LABEL instead of UUID until kiwi 9.19.0, so tw and Leap 15.2+ are fine
     verify_mounts unless is_leap('<15.2') && is_aarch64;
+    assert_script_run 'cat /proc/cmdline';
```
Comment (ggardet) [inline tests/jeos/firstrun.pm:169]: Why do you need this?
If you want to store the info, it is probably better to use `record_info` or upload the result to a file?

Comment (mloviska) [reply tests/jeos/firstrun.pm:169]: It was only for my debugging purposes, so I could easily compare the kernel parameters. I will remove it, thanks! :)

File: lib/autofs_utils.pm [hunk 2]
```diff
@@ -134,6 +134,11 @@ sub configure_service {
         test_conf_file_content => $test_conf_file_content,
         test_mount_dir         => $test_mount_dir);
 
+    if (script_run('test -f /etc/nsswitch.conf') && !script_run('rpm -q glibc')) {
```
Comment (Vogtinator) [inline lib/autofs_utils.pm:137]: In which case would `rpm -q glibc` fail?

File: lib/main_common.pm [hunk 3]
```diff
@@ -577,7 +577,7 @@ sub load_jeos_tests {
         unless (get_var('INSTALL_LTP')) {
             loadtest "console/force_scheduled_tasks";
             loadtest "jeos/grub2_gfxmode";
-            loadtest 'jeos/revive_xen_domain' if check_var('VIRSH_VMM_FAMILY', 'xen');
+            loadtest 'jeos/revive_xen_domain' if (check_var('VIRSH_VMM_FAMILY', 'xen') && (get_var('EXTRATEST') !~ m/filesystem/));
```
Comment (Vogtinator) [inline lib/main_common.pm:580]: What is this test actually needed for?

**Review by Vogtinator — COMMENTED** [review]
AFAICT this PR is mixing four independent changes?

File: lib/autofs_utils.pm (hunk 2 above)
Comment (mloviska) [reply lib/autofs_utils.pm:137]: Oops, I meant `glibc-devel`. Thanks!

Comment (mloviska) [reply lib/autofs_utils.pm:137]: but can be skipped anyway

File: lib/autofs_utils.pm [hunk 4]
```diff
@@ -134,6 +134,11 @@ sub configure_service {
...
+        record_soft_failure('boo#20200812 openQA test fails in autofs - glibc-devel installs /usr/include/netdb.h which wrongly defines _PATH_NSSWITCH_CONF="/etc/nsswitch.conf" (/etc/nsswitch.conf was moved to /usr/etc/nsswitch.conf)');
```
Comment (Vogtinator) [inline lib/autofs_utils.pm:138]: I don't understand this - the bugref seems to be wrong and what impact would `glibc-devel` have here?

File: lib/main_common.pm (hunk 3 above)
Comment (mloviska) [reply lib/main_common.pm:580]: Not really sure, but I wouldn't like to decommission this test case in this PR. It is used in most XEN tests. In the filesystems test suite we do not really want to reboot before `snapper_undochange` as the kernel parameters introduced by `bootloader_svirt.pm` would vanish.

Comment (Vogtinator) [reply lib/main_common.pm:580]: That it passes without this test in the xen + filesystem case proves that it's not neccessary though, doesn't it?

File: lib/autofs_utils.pm (hunk 4 above)
Comment (mloviska) [reply lib/autofs_utils.pm:138]: > AFAICT this PR is mixing four independent changes?

Yeah, maybe I should split the changes in more commits.

Comment (mloviska) [reply lib/autofs_utils.pm:138]: > I don't understand this - the bugref seems to be wrong and what impact would `glibc-devel` have here?

`glibc-devel` owns `/usr/include/netdb.h` which is used by autofs to pick up the nsswitch configuration file. `/usr/include/netdb.h` points to a file, which has been moved. As there is no solution yet and the discussion continues in the bugref, I have rather included it here.

Comment (Vogtinator) [reply lib/autofs_utils.pm:138]: Does it actually parse a C header file during runtime? I don't see any dependency here.

I assume it's a compile time thing, so the test for `glibc-devel` is a red herring.

Comment (mloviska) [reply lib/autofs_utils.pm:138]: > I assume it's a compile time thing, so the test for `glibc-devel` is a red herring.

Agreed, removed from the test case.

File: lib/main_common.pm (hunk 3 above)
Comment (mloviska) [reply lib/main_common.pm:580]: Ok, so let me try also `xen-hvm` + fs test suite without it. :)

Comment (mloviska) [reply lib/main_common.pm]: Seems to be fine http://kepler.suse.cz/tests/1654#

Comment (Vogtinator) [reply lib/main_common.pm]: Ok, so +1 for removing `revive_xen_domain`

File: lib/main_common.pm [hunk 5]
```diff
@@ -1805,16 +1805,14 @@ sub load_extra_tests_filesystem {
...
-    loadtest 'console/snapper_create';
     if (get_var('NUMDISKS', 0) > 1 && (is_sle('12-sp3+') || is_leap('42.3+') || is_tumbleweed)) {
         # On JeOS we use kernel-defaul-base and it does not have 'dm-thin-pool'
         # kernel module required by thin-LVM
         loadtest 'console/snapper_thin_lvm' unless is_jeos;
     }
     loadtest 'console/snapper_used_space' if (is_sle('15-SP1+') || (is_opensuse && !is_leap('<15.1')));
     loadtest "console/udisks2" unless is_sle;
-    loadtest "console/zfs" if (is_leap(">=15.1") && is_x86_64);
+    loadtest "console/zfs" if (is_leap(">=15.1") && is_x86_64 && !is_jeos);
```
Comment (pdostal) [inline lib/main_common.pm:1815]: Maybe @grisu48 would like to look at this later.

**Review by pdostal — APPROVED** [review]

File: lib/autofs_utils.pm (hunk 4 above)
Comment (DimStar77) [reply lib/autofs_utils.pm:138]: > I don't understand this - the bugref seems to be wrong and what impact would `glibc-devel` have here?

The bug ref was unfortunately missed to be fixed; this bug ref is invalid

**Pull request merged**